pkg/pinentry/pinentry.go

```diff
@@ -125,13 +126,15 @@ func (c *Client) GetPin() ([]byte, error) {
 	if bytes.HasPrefix(buf, []byte("OK")) {
 		return nil, nil
 	}
-	if bytes.HasPrefix(buf, []byte("S PASSWORD_FROM_CACHE")) {
```

Comment:
Not sure why you'd want to have a for loop instead of the more specific if statement here.
https://git.gnupg.org/cgi-bin/gitweb.cgi?p=pinentry.git;a=blob;f=doc/pinentry.texi;h=a4acd8b9f27bf2cdbd50aead2757872978bbe9f6;hb=HEAD#l483

Reply:
Since I'm not sure how many how many "headers" are being returned.

Maybe I'm reading the spec wrong, but this should work either way?

Reply:
Yeah I think it should, but it would misinterpret a status message "S PIN_REPEATED"

Reply:
Unless we want to do the same in case of such a status, actually. I haven't dug too much into how pinentry works